### DISCUSSION TITLE
Add a way to have `ObservableProperty` implicit over all fields in a class

### DISCUSSION THREAD

**Discussion by TheBlubb14** [CommunityToolkit/dotnet#402] [Ideas]
Currently my code looks like this

```cs
partial class Device : ObservableObject
{
    [ObservableProperty]
    private bool isConnected;

    [ObservableProperty]
    private string name;

    [ObservableProperty]
    private int id;

    private int key;
}
```

but I would like to don't have to put the `ObservablePropertyAttribute` above every field in `Device`.
I would like to have a new attribute called something like 

`ObservableClassAttribute`
```cs
[AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = false)]
public sealed class ObservableClassAttribute : Attribute
{
}
```
`IgnoreObservablePropertyAttribute `
```cs
[AttributeUsage(AttributeTargets.Field, AllowMultiple = false, Inherited = false)]
public sealed class IgnoreObservablePropertyAttribute : Attribute
{
}
```

and use it like this

```cs
[ObservableClass]
partial class Device : ObservableObject
{
    private bool isConnected;
    private string name;
    private int id;

    [IgnoreObservableProperty]
    private int key;
}
```

This has some similarities #374 but I dont need to have properties from my code, they can be autogenerated.

**TLDR**: Generate observable property for all fields inside class and manually exlude fields with an ignore-attribute

**Comment by tradegear** [comment]
This sure makes a lot of sense.
